avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroTypeDeserializer.java

```diff
@@ -57,6 +54,13 @@ public Object deserializeTypedFromScalar(JsonParser p, DeserializationContext ct
 
     @Override
     public Object deserializeTypedFromAny(JsonParser p, DeserializationContext ctxt) throws IOException {
+        AnnotationIntrospector ai = ctxt.getAnnotationIntrospector();
+        if (ai != null && _property != null) {
+            JsonDeserializer<?> deser = ctxt.deserializerInstance(_property.getMember(), ai.findDeserializer(_property.getMember()));
```

Comment:
This is extremely inefficient as well; usually deserializers are cached, and if this is not done much (most?) of the time is spent on fetching and contextualizing deserializer...